packages/editor/src/components/provider/content-only-lock-sections.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch, useRegistry } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { useEffect } from '@wordpress/element';
+import { store as blocksStore } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+/**
+ * Component that when rendered, makes it so that the editor allows only
+ * specific sections to be edited in content-only mode for templates.
+ */
+export default function ContentOnlyLockSections() {
+	const registry = useRegistry();
+
+	const { setBlockEditingMode, unsetBlockEditingMode } =
+		useDispatch( blockEditorStore );
+
+	const selected = useSelect( ( select ) => {
+		const {
+			getBlockOrder,
+			getClientIdsOfDescendants,
+			getClientIdsWithDescendants,
+		} = select( blockEditorStore );
+		const { getEditorSettings } = select( editorStore );
+		const { sectionRootClientId } = getEditorSettings();
```

Comment:
I'm finding this value is incorrect on the homepage of TT4. It's likely due to the `template-locked` being applied which causes this conditional to find the "wrong" sections:

https://github.com/WordPress/gutenberg/blob/3aaf6ffeaa433b61b44c9bbd7ea8c86bb773337b/packages/editor/src/components/provider/use-block-editor-settings.js#L144-L146

What we need in this case is for it to use the `<main>` algorithm.

Reply:
I think this is now fixed by some commits I made.